File: sdk/csharp/src/IO.StockX.Examples/PlaceNewLowestAskExample.cs
using System;
using System.Diagnostics;
using IO.StockX.Api;
using IO.StockX.Client;
using IO.StockX.Model;

namespace Example
{
    /**
    * <p>Demonstrates creating a new lower asking price for a product as a seller.</p>
    */
    public class PlaceNewLowestAskExample
    {
        static public void Main()
        {
            // Configure API key authorization: api_key
            Configuration.Default.AddApiKey("x-api-key", ExampleConstants.AWS_API_KEY);

            var stockx = new StockXApi();
                stockx.Configuration.BasePath = ExampleConstants.STOCKX_ENDPOINT;
            
            var login = new LoginRequest(ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);
            
            try
            {
                // Login and fetch the jwt header for authentication use in the request
                ApiResponse<LoginResponse> result = stockx.LoginWithHttpInfo(login);
                var jwt = result.Headers[ExampleConstants.JWT_HEADER];

                stockx.Configuration.DefaultHeader["jwt-authorization"] = jwt;
                
                // Find a product with search
                var search = stockx.Search("Jordan Retro Black Cat");

                // Lookup the product's data by its search id
                var firstResultStyle = search.Hits[0].StyleId;

                var productInfo = stockx.LookupProduct(firstResultStyle, "11");

                // Get the market data (highest bids, lowest asks, etc) about the product
                var id = productInfo.Data[0].Id;
                var productUuid = productInfo.Data[0].Attributes.ProductUuid;

                var marketData = stockx.GetProductMarketData(id, productUuid);

                // Get the lowest ask and increment it
                var lowestAsk = marketData.Market.LowestAsk;
                lowestAsk++;

                // Create a portfolio item request with a higher bid
                var item = new PortfolioRequestPortfolioItem();
                    item.Amount = lowestAsk.ToString();
                    item.SkuUuid = "bae25b67-a721-4f57-ad5a-79973c7d0a5c";
                    item.MatchedWithDate = "2018-12-12T05:00:00+0000";
                    item.ExpiresAt = "2018-12-12T12:39:07+00:00";

                var request = new PortfolioRequest();
                    request.PortfolioItem = item;
                    request.Customer = result.Data.Customer;
                    request.Timezone = "America/Detroit";

                // Submit the ask
                var askResp = stockx.NewPortfolioAsk(request);

                Console.WriteLine(askResp);
            }
            catch (Exception e)
            {
            	Console.WriteLine(e.ToString());
                Console.WriteLine("Exception when creating new lowest ask: " + e.Message );
            }
        }
    }
}

File: sdk/python/src/place_new_lowest_ask_example.py
from __future__ import print_function

import time
import io_stockx
from example_constants import ExampleConstants
from io_stockx.rest import ApiException
from pprint import pprint

# Configure API key authorization: api_key
configuration = io_stockx.Configuration()

configuration.host = "https://gateway.stockx.com/stage"
configuration.api_key['x-api-key'] = ExampleConstants.AWS_API_KEY

# create an instance of the API class
stockx = io_stockx.StockXApi(io_stockx.ApiClient(configuration))
login = io_stockx.LoginRequest(email=ExampleConstants.STOCKX_USERNAME, password=ExampleConstants.STOCKX_PASSWORD)

try:
    # Attempts to log the user in with a username and password.
    api_response = stockx.login_with_http_info(login)

    # Get the customer object after login
    customer = api_response[0]

    # Get the login's assigned jwt token
    jwt_token = api_response[2]['Jwt-Authorization']

    # Use the jwt token to authenticate future requests
    stockx.api_client.set_default_header('jwt-authorization', jwt_token)

    # Search for a type of product
    search_result = stockx.search('Jordan Retro Black Cat')

    first_hit = search_result.hits[0]
    style_id = first_hit.style_id

    # Lookup the first product returned from the search
    product = stockx.lookup_product(identifier=style_id, size='11')

    # Get the current market data for the product (highest bid info, etc.)
    attributes = product.data[0].attributes
    id = product.data[0].id
    uuid = attributes.product_uuid

    # Get the product market data
    market_data = stockx.get_product_market_data(id, sku=uuid)

    # Get the lowest ask for the product and decrement it
    lowest_ask = market_data.market.lowest_ask
    lowest_ask += 1

    # Create a portfolio item request with a higher bid
    item = io_stockx.PortfolioRequestPortfolioItem()

    item.amount = lowest_ask
    item.sku_uuid = "bae25b67-a721-4f57-ad5a-79973c7d0a5c"
    item.matched_with_date = "2018-12-12T05:00:00+0000"
    item.expires_at = "2018-12-12T12:39:07+00:00"

    request = io_stockx.PortfolioRequest()
    request.portfolio_item = item
    request.customer = customer
    request.timezone = "America/Detroit"

    # Submit the ask
    ask_resp = stockx.new_portfolio_ask(request)

    pprint(ask_resp)
except ApiException as e:
    print("Exception when calling StockXApi->new_portfolio_ask: %s\n" % e)
